connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -396,6 +396,8 @@ private void commitOffsets(long now, boolean closing) {
                     log.debug("{} Rewinding topic partition {} to offset {}", this, entry.getKey(), entry.getValue().offset());
                     consumer.seek(entry.getKey(), entry.getValue().offset());
                 }
+                // Clear messages that might have been left in messageBatch after last poll
+                messageBatch.clear();
```

Comment:
Do we need to also adjust `pausedForRedelivery` and resume the topic partitions here?